src/main/java/com/uber/jenkins/phabricator/HarbormasterConduitNotifier.java

```diff
@@ -0,0 +1,114 @@
+import org.apache.http.NameValuePair;
+import org.apache.http.client.ClientProtocolException;
+import org.apache.http.client.HttpResponseException;
+import org.apache.http.client.entity.UrlEncodedFormEntity;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClientBuilder;
+import org.apache.http.message.BasicNameValuePair;
+import org.kohsuke.stapler.DataBoundConstructor;
+
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.util.ArrayList;
+import java.util.List;
+
+public class HarbormasterConduitNotifier extends Notifier {
+    private static final String TAG = "harbormaster-conduit";
+
+    private final String buildTarget;
+
+    @DataBoundConstructor
+    public HarbormasterConduitNotifier(String buildTarget) {
+        this.buildTarget = buildTarget;
+    }
+
+    public BuildStepMonitor getRequiredMonitorService() {
+        return BuildStepMonitor.NONE;
+    }
+
+    @Override
+    public final boolean perform(final AbstractBuild<?, ?> build, final Launcher launcher,
+                                 final BuildListener listener) throws InterruptedException, IOException {
+        String harbormasterResult = "fail";
+        EnvVars environment = build.getEnvironment(listener);
+        Logger log = new Logger(listener.getLogger());
+        Result buildResult = build.getResult();
+
+        // TODO(cburroughs): make this more sophisticated, or configurable
+        if (buildResult.isBetterOrEqualTo(hudson.model.Result.UNSTABLE)) {
+            harbormasterResult = "pass";
+        }
+        log.info(TAG, "Translated harbormaster build result: " + harbormasterResult);
+        String buildTargetPHID = environment.get(buildTarget);
+        if (buildTargetPHID == null) {
+            log.info(TAG, "Unable to find value for paramter: " + buildTarget);
+            return false;
+        }
+        harbormasterSendMessage(log, getDescriptor().getConduitURL(), getDescriptor().getConduitToken(), buildTargetPHID, harbormasterResult);
+        return true;
+    }
+
+    private void harbormasterSendMessage(Logger log, String conduitURL, String conduitToken, String buildTargetPHID, String harbormasterResult) {
+        try {
+            CloseableHttpClient client = HttpClientBuilder.create().build();
+            HttpPost httppost = new HttpPost(conduitURL + "/harbormaster.sendmessage");
+            List<NameValuePair> formparams = new ArrayList<NameValuePair>();
+            formparams.add(new BasicNameValuePair("api.token", conduitToken));
+            formparams.add(new BasicNameValuePair("buildTargetPHID", buildTargetPHID));
+            formparams.add(new BasicNameValuePair("type", harbormasterResult));
+            UrlEncodedFormEntity entity = new UrlEncodedFormEntity(formparams);
+            httppost.setEntity(entity);
+            HttpResponse response = client.execute(httppost);
```

Comment:
Oh, I see the intention here (not using arcanist). Yes, this is good, but we should refactor the existing sendmessage/postcomment to also use a Harbormaster class (extracted from this notifier) instead of running `arc call-conduit`.